### ISSUE TITLE
Add Labor Report to the Barn Kit

### ISSUE THREAD

**Issue by braughtg** [DickinsonCollege/FarmData2#419]
A report can be generated showing the amount of labor performed on tasks.

The functionality of the FarmData2 Labor Report will be similar to the Labor Report in the original FarmData, which can be seen here:

https://farmdata.dickinson.edu/Labor/laborReport.php?tab=labor:labor_report

In FarmData2 this Labor Report will appear in the Barn Kit tab and will allow the user to generate a report that displays information about the requested labor events. When generating and displaying this report, the style and user experience should be similar to:
* The examples that appear in the FD2 Example tabs
  * Their defining code is in the html files in the directory: farmdata2_modules/fd2_tabs/fd2_example
* The Seeding Report form that appears in the Barn Kit.
  * Its defining code is in the file: farmdata2_modules/fd2_tabs/fd2_barn_kit/seedingReport.html

Data Model Notes:
* All of the different types of Logs have a "quantity" field that contains labor information about the action represented by the log (e.g. direct seeding, harvesting, etc).

Information about the Tray Seeding Logs, Transplanting Logs, Harvest Logs and Planting Assets:
* General documentation for the sample database: https://github.com/DickinsonCollege/FarmData2/tree/main/docker/sampleDB
  * API Calls in that document show how to retrieve existing logs and assets that can be used to understand the JSON structures and their contents.
* The JSON Structures used to create the Logs and Assets in the sample database can be found in the python files in the docker/sampleDB directory:
  * The JSON structure for a Tray Seeding Log and its associated Planting Asset can be found in the file: docker/sampleDB/addTraySeedings.py
  * The JSON structure for a Direct Seeding Log and its associated Planting Asset can be found in the file: docker/sampleDB/addDirectSeedings.py
  * The JSON structure for a Transplanting Log can be found in the file: docker/sampleDB/addTransplantings.py
  * The JSON structure for a Harvest Log can be found in the file: docker/sampleDB/addHarvests.py
* Information about the API calls used to create Logs and Assets using the JSON structures can be found:
  * In the file: docker/sampleDB/utils.py
  * In section of CONTRIBUTING.md that points to the FarmOS API documentation.

**Comment by johnmaccormick** [comment]
If you get a login error at the first link above, you need to first visit [https://farmdata.dickinson.edu/guest.php](https://farmdata.dickinson.edu/guest.php) to log in as a guest.

**Comment by astern21** [comment]
Is there need for a separate issue to input labor? or will there already be labor information when other input logs?

**Comment by braughtg** [comment]
> Is there need for a separate issue to input labor? or will there already be labor information when other input logs?

All of the actions that require labor are represented by logs in the FarmOS/FarmData2.  Thus, the input of labor (time and people) is done when those logs are created (e.g. See the Seeding Input form in the FieldKit).  This report displays the labor information from the existing logs.  That said, the labor amounts that are displayed should be editable here in case an error is discovered.  So the table used here should be similar to the Seeding Report in the BarnKit with edit buttons for the rows.  Though my current thought is to not allow deletion from this report as it seems inconsistent to delete seedings or harvests or transplantings from the labor report.  The individual reports for those events allow logs to be deleted.

**Comment by astern21** [comment]
**Current status of the issue:**

The base of the Labor Report is a copy of the Seeding report html source file as the functionality and UX should be similar. We removed the column with the option to delete the log and also removed the function from the html file since we should not be able to delete a log in the backend (STORY 1). Because the labor report source file is a copy of the seeding report file, the date range selection(STORY 3) and edit button (STORY 5) were able to be completed with the given code.  Right now we have a dropdown called task which is currently a copy of the area dropdown(STORY 8). We are still figuring out how to get the api call to populate the dropdown with types of tasks instead of areas so that users are able to filter the generated report by the specific task. We also are currently working on adding a Task column in the generated table (STORY 1). We are still unable to do this. The table is generated based on customization of visible and invisible columns. We tried adding a header for task column in the headers Vue element and set its visibility to true, the header appears In the table however, all the headers after the task column shifted to the right but its contents stayed the same with the last column in the table remaining empty. In other words, the headers and contents of the column do not align.  We have to figure out someway to add a column for the type of task of the log where the content of the table matches the header. We first need to add the column then later populate it with an api call, and we do not want the task column to be the very last column in the table.
